File: src/python/finished/minimum_equal_sum_of_two_rrays_after_replacing_zeros.py
from typing import List

# https://leetcode.com/problems/minimum-equal-sum-of-two-arrays-after-replacing-zeros


class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        ltotal, lzero = 0, 0
        rtotal, rzero = 0, 0

        for n in nums1:
            if n == 0:
                lzero += 1
            ltotal += n

        for n in nums2:
            if n == 0:
                rzero += 1
            rtotal += n

        lmin = ltotal + lzero
        rmin = rtotal + rzero

        if lmin < rmin and lzero == 0:
            return -1

        if rmin < lmin and rzero == 0:
            return -1

        return max(lmin, rmin)

File: src/cpp/finished/minimum_equal_sum_of_two_rrays_after_replacing_zeros.cpp
#include <vector>

using namespace std;

// https://leetcode.com/problems/minimum-equal-sum-of-two-arrays-after-replacing-zeros

class Solution {
 public:
  long long minSum(vector<int>& nums1, vector<int>& nums2) {
    long long ltotal = 0;
    long long rtotal = 0;
    long long lzero = 0;
    long long rzero = 0;

    for (const auto n : nums1) {
      ltotal += n;
      if (n == 0) {
        lzero++;
      }
    }

    for (const auto n : nums2) {
      rtotal += n;
      if (n == 0) {
        rzero++;
      }
    }

    auto lmin = ltotal + lzero;
    auto rmin = rtotal + rzero;

    if (lmin < rmin && lzero == 0) {
      return -1;
    }

    if (rmin < lmin && rzero == 0) {
      return -1;
    }

    return lmin > rmin ? lmin : rmin;
  }
};